src/commands/serve.rs

```diff
@@ -386,7 +386,7 @@ async fn handle_request(
     let (sender, receiver) = tokio::sync::oneshot::channel();
 
     // TODO: need to track the join handle, but don't want to block the response on it
```

Comment:
@elliottt, I guess if we adopt this PR we can remove this TODO line?

Reply:
Yes, please remove this TODO and we can land this PR